src/relationUtils.ts

```diff
@@ -276,3 +280,117 @@ export function toJson(output: Relation[] | ArrowRelation[]): any {
 
   return result;
 }
+
+export function filterRelations(relations: ArrowRelation[], keys: string[]) {
+  const len = keys.length;
+  const keysStr = keys.join('/');
+  const filteredRelations: ArrowRelation[] = [];
+
+  relations.forEach(r => {
+    const relationKeys = getKeys(r.relationId);
+
+    if (relationKeys.slice(0, len).join('/') === keysStr) {
+      const newKeys = relationKeys.slice(len);
+
+      filteredRelations.push({
+        ...r,
+        relationId: getRelationId(newKeys),
+      });
+    }
+  });
+
+  return filteredRelations;
+}
+
+export type DiagnosticPos = {
+  line: number;
+  character: number;
+};
+
+export type DiagnosticRange = {
+  start: DiagnosticPos;
+  end: DiagnosticPos;
+};
+
+export type Diagnostic = {
+  range: DiagnosticRange[];
+  message: string;
+  severity: string;
+  code: string;
```

Comment:
diagnostics can also have a `model` (equivalent to the current `path` in a problem), but it is only present if the problem is not in a query (i.e. inserted into an edb/installed)

Reply:
thanks!